packages/medusa/src/repositories/discount-condition.ts

```diff
@@ -35,6 +36,17 @@ type DiscountConditionResourceType = EntityTarget<
 
 @EntityRepository(DiscountCondition)
 export class DiscountConditionRepository extends Repository<DiscountCondition> {
+  async findOneWithDiscount(
```

Comment:
should we instead use a left join here and then in the retrieve method we can check that discount condition and that the discount exists to? that way in the remove endpoint we can load the discounConditionRepo and use that method instead of retrieving from the discountService and that way we ensure that the discount exists or throw and that the condition belongs to the requested discount. wdyt?